src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/services/operation/ReplicationCheckerTest.java

```diff
@@ -0,0 +1,245 @@
+
+    @Mock
+    private CFTTaskDatabaseService cftTaskDatabaseService;
+
+    @Mock
+    private MIReportingService miReportingService;
+
+    @InjectMocks
+    private ReplicationChecker replicationChecker;
+
+    private final TaskOperationRequest request = new TaskOperationRequest(
+        TaskOperation.builder()
+            .type(TaskOperationType.PERFORM_REPLICATION_CHECK).build(),
+        List.of()
+    );
+
+    @Test
+    void should_process_replication_checker_operation() {
+        Map<String, Object> resourceMap = replicationChecker.performOperation(request).getResponseMap();
+        List<?> tasks = (List<?>) resourceMap.get("replicationCheckedTaskIds");
+        assertEquals(0, tasks.size());
+        List<?> notReplicated = (List<?>) resourceMap.get("notReplicatedTaskIds");
+        assertEquals(0, notReplicated.size());
+        verify(cftTaskDatabaseService, times(1)).findLastFiveUpdatedTasks();
+    }
+
+    @Test
+    void should_process_replication_checker_operation_for_empty_list() {
+        when(cftTaskDatabaseService.findLastFiveUpdatedTasks()).thenReturn(List.of());
+
+        Map<String, Object> resourceMap = replicationChecker.performOperation(request).getResponseMap();
+        List<?> tasks = (List<?>) resourceMap.get("replicationCheckedTaskIds");
+        assertEquals(0, tasks.size());
+        List<?> notReplicated = (List<?>) resourceMap.get("notReplicatedTaskIds");
+        assertEquals(0, notReplicated.size());
+        verify(cftTaskDatabaseService, times(1)).findLastFiveUpdatedTasks();
+    }
+
+    @Test
+    void should_process_replication_checker_operation_for_found_history() {
+        TaskResource resource = new TaskResource("1",
+            OffsetDateTime.parse("2021-05-09T20:15:50.345875+01:00"), "claim", "someuser");
+        TaskHistoryResource history = new TaskHistoryResource("1",
+            OffsetDateTime.parse("2021-05-09T20:15:50.345875+01:00"), "claim", "someuser");
+        when(cftTaskDatabaseService.findLastFiveUpdatedTasks()).thenReturn(List.of(resource));
+        when(miReportingService.findByTaskIdOrderByLatestUpdate("1")).thenReturn(List.of(history));
+
+        Map<String, Object> resourceMap = replicationChecker.performOperation(request).getResponseMap();
+        List<?> tasks = (List<?>) resourceMap.get("replicationCheckedTaskIds");
+        assertEquals(1, tasks.size());
+        List<?> notReplicated = (List<?>) resourceMap.get("notReplicatedTaskIds");
+        assertEquals(0, notReplicated.size());
+
+        verify(cftTaskDatabaseService, times(1)).findLastFiveUpdatedTasks();
+
+        await()
+            .pollInterval(1, SECONDS)
+            .atMost(20, SECONDS)
+            .until(() -> {
+                verify(miReportingService, times(1)).findByTaskIdOrderByLatestUpdate("1");
+                return true;
+            });
+
+    }
+
+    @Test
+    void should_process_replication_checker_operation_for_not_found_history_timestamp_mismatch() {
```

Comment:
We can use CapturedOutput to verify the replicationLog is invoked for this scenario.

Reply:
Added log assertion for all error scenario tests